app/src/main/java/com/android/sample/ui/activitydetails/ActivityDetailsScreen.kt

```diff
@@ -490,3 +495,33 @@ fun CommentItem(
     }
   }
 }
+
+@Composable
+fun LikeButton(profile: User?, activity: Activity?, profileViewModel: ProfileViewModel) {
+  var isLiked by remember {
+    mutableStateOf(activity?.let { profile?.likedActivities?.contains(it.uid) } ?: false)
+  }
+
+  if (profile != null) {
+    IconButton(
+        modifier = Modifier.testTag("likeButton$isLiked"),
+        onClick = {
+          isLiked = !isLiked
+          if (isLiked) {
+            if (activity != null) {
```

Comment:
The nested "if condition" and repetition here is not necessary, you could be using the following syntax:
activity?.let { activity -> ...} which will directly ensure activity is not null